### DISCUSSION TITLE
How to forward boolean flag

### DISCUSSION THREAD

**Discussion by avkonst** [nushell/nushell#6828] [Q&A]
see related https://github.com/nushell/nushell/issues/6827
I am looking for a workaround this problem.
How can I make sure that the following prints true or false depending on how the script is invoked? Currently it always prints true.

```
def cmd  [--flag: bool] {
    echo $flag
}
def main [--flag: bool] {
    cmd --flag=$flag
}
```

**Comment by fdncred** [comment]
How about this.
```
> def cmd [--flag(-f): bool] { print $flag }
> cmd
false
> cmd -f
true
> cmd --flag
true
```

**Comment by avkonst** [comment]
I understand it is how you call cmd. I need the implementation of main which does not have if over cmd calling with different params, because my real example has lots of bool flags and I can not write all possible combinations calling cmd in different ways.

**Reply by fdncred** [reply]
The point I was trying to get across is that I believe you call it like `cmd --flag` and not `cmd --flag=false` and if you pass `--flag` that means true and if you don't pass `--flag` then it's false.

We could support --flag=false somewhere. that sounds vaguely familiar.

**Comment by avkonst** [comment]
I understand what you are saying but now try to apply it to the implementation in main function. How would it look like?

**Reply by fdncred** [reply]
I created this file called `test.nu` and this is what's in it.
```
def main [--flag(-f): bool] {
    echo $flag
}
```
If I want to run that script, I can do `nu test.nu` and it will return `false`. If I want to supply the flag, I can do `nu test.nu -f` and it will return `true`.
![image](https://user-images.githubusercontent.com/343840/197534542-fde87691-13b9-48d3-a575-325e4e4708ce.png)

**Comment by avkonst** [comment]
I understand all of this. The question was different: how to forward this flag from main to cmd function? See my original question in the top. In other words, how would you rewrite your main function by calling my cmd without using if statement inside of the main function

**Reply by fdncred** [reply]
I don't think there is another way other than having if statements.

This works for me, which is testing for the possibility of each flag
```
def cmd  [--flag(-f): bool] {
    echo $flag
}
def main [--flag(-f): bool] {
    if $flag {
        cmd --flag
    } else {
        cmd
    }
}
```
```
> nu test.nu -f
true
> nu test.nu
false
```

**Comment by avkonst** [comment]
Sure I used this. And now let's say there are 3 boolean arguments you need to forward, what a messy code would it be?

**Reply by fdncred** [reply]
Indeed. I like your original solution in the question. Too bad it doesn't work right now.

**Comment by theAkito** [comment]
This works for me.

```nu
def cmd  [--flag = false] {
    echo $flag
}

def main [--flag = false] {
    cmd --flag=$flag
}
```

**Comment by avkonst** [comment]
This code compiles but does not produce the correct result. It always
prints true.

**Reply by theAkito** [reply]
No. You are incorrect.

**Reply by theAkito** [reply]
Which version of Nu are you using?

**Reply by theAkito** [reply]
```bash
$ nu booltest.nu --flag=true
true
$ nu booltest.nu --flag=false
false
```

Literally taken from my terminal. I just stripped user & hostname.

**Comment by avkonst** [comment]
1 month since installed the latest

**Reply by theAkito** [reply]
Which one would that be, when you issue `nu --version`?

**Reply by theAkito** [reply]
https://github.com/nushell/nushell/releases/tag/0.70.0 This is the newest version.

**Comment by avkonst** [comment]
0.69.1

**Reply by theAkito** [reply]
So, maybe update?
